### PR TITLE
Multi-LRAUV Swimming Race Example

### PR DESCRIPTION
**Pull request by mabelzhang**
Signed-off-by: Mabel Zhang <mabel@openrobotics.org>

# 🎉 New feature

## Summary

An example world and controller for a multi-LRAUV swimming race in an underwater environment. Demonstrates how to control the basic set of plugins using a random controller.

Race results vary in every run. Don't take it personally if your favorite LRAUV doesn't win. (Disclosure: they are all the same! Except for their names -- uh oh, shouldn't have said that.)

## Remarks

@arjo129, two things I noticed:
1. `ThrusterPlugin` takes a negative value for going forward. Is that the convention?
   I noticed in the LRAUV model.sdf that the `propeller_joint` is set to negative effort and velocity:
   ```
          <effort>-1</effort>
          <velocity>-1</velocity>
   ```
   Is the actual robot propeller just designed to go forward when rotated in that direction?

2. The robots are going a lot faster here than in our downstream version. All the parameters are the same, and the command values are in the same bounds (other than the thrust being (-31.4 rad/s, 0) as opposed to (0, +31.4 rad/s). Is something different here (did hydrodynamics get faster)?

## Test it

Terminal 1:
```
cd examples/worlds/
ign gazebo -r multi_lrauv_race.sdf
```

Terminal 2:
```
cd examples/standalone/multi_lrauv_race
mkdir build && cd build
cmake ..
make
./multi_lrauv_race
```

You should see something like this:

https://user-images.githubusercontent.com/7608908/120051799-23e1f800-bff0-11eb-894f-295abb4589e7.mp4

## Checklist
- [x] Signed all commits for DCO
- [ ] ~~Added tests~~
- [x] Added example and/or tutorial
- [x] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

### CODE REVIEW

**Review by arjo129 — CHANGES_REQUESTED** [review]
It works for me. If you're curious: Daphne won :trophy:

I've left a comment as to why we seem to be moving so quickly in the review. It has to do with units. Once that is adressed the environment looks good to me.

As for the thrust taking a negative plugin, this is not due to any convention.  To be fair, there are both clockwise and counterclockwise thrusters out there in the wild. As for the effort and velocity being negative, I'm not too sure why they are negative and if this has any impact on the thruster direction at all as the thruster plugin does not read those values.

File: examples/standalone/multi_lrauv_race/multi_lrauv_race.cc [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+// Fin joint limits from tethys model.sdf
+double random_angle_within_limits(double min=-0.261799, double max=0.261799)
+{
+  return min + static_cast<float>(rand()) /
+    (static_cast<float>(RAND_MAX / (max - min)));
+}
+
+// Nominal speed is thruster 300 rpm = 31.4 radians per second
+double random_thrust_within_limits(double min=-31.4, double max=31.4)
```
[the remark this answers is not in the document]
Comment (mabelzhang) [reply examples/standalone/multi_lrauv_race/multi_lrauv_race.cc]: Cool, thanks! It looks slower now. 585f31e

**Comment by mabelzhang** [comment]
> As for the effort and velocity being negative, I'm not too sure why they are negative and if this has any impact on the thruster direction at all as the thruster plugin does not read those values.

I should have checked the specs. -1 is actually the default value, to indicate no max is to be enforced. The tags are for max effort and limit. http://sdformat.org/spec?ver=1.8&elem=joint#limit_effort
So no questions there anymore.

> As for the thrust taking a negative plugin, this is not due to any convention. To be fair, there are both clockwise and counterclockwise thrusters out there in the wild.

I wonder if it could be because the current Tethys model has x pointing to the stern. Maybe after we flip it to have x pointing to the bow in some weeks, it'll go forward with positive thrust.

I'm mainly coming from the rationale that if the user does the most vanilla thing with their joint axes and follow the convention (x forward), then when they try out the Thruster plugin, it should do the most intuitive thing, which to me is positive going forward. In this case, I think the model we're using is not following conventions, so that's a problem with the model, not the plugin. Let's see in a few weeks if this is confirmed!

**Review by arjo129 — APPROVED** [review]
This works as described for me. This time both Daphne and Triton were tied :trophy: .

**Comment by mabelzhang** [comment]
Cool! Daphne wins more for you too huh, she won the most for me in the beginning but later spread out better, I think. I'm gonna rerun CI because this looks too red even for an example that shouldn't affect the tests...

**Comment by mabelzhang** [comment]
@osrf-jenkins run tests

**Comment by mabelzhang** [comment]
Double-checked that the failed Ubuntu tests are flaky. Actually all OSes' CI have flaky tests and none are required apparently :sweat_smile:
Merging.

**Pull request merged**